### PR TITLE
Emit processing time metric

### PR DESCRIPTION
**Pull request by mauroservienti**
- depends on https://github.com/Particular/NServiceBus/pull/7077
- depends on https://github.com/Particular/NServiceBus/pull/7058

This PR introduces an OpenTelemetry-native metric for measuring the message's [processing time](https://docs.particular.net/monitoring/metrics/definitions#metrics-captured-processing-time), which include the pipeline.

POA:
- [x] Update [public docs](https://docs.particular.net/nservicebus/operations/opentelemetry#meters-emitted-meters)

### CODE REVIEW

File: src/NServiceBus.Core/OpenTelemetry/Metrics/MessagingMetricsFeature.cs [hunk 1]
```diff
@@ -22,6 +24,16 @@ protected internal override void Setup(FeatureConfigurationContext context)
...
+        context.Pipeline.OnReceivePipelineCompleted((pipeline, token) =>
+        {
+            List<Task> tasks =
+            [
+                processingTimeMetrics.Record(pipeline, token),
+                criticalTimeMetrics.Record(pipeline, token)
+            ];
+
+            return Task.WhenAll(tasks);
+        });
```
Comment (DavidBoike) [inline src/NServiceBus.Core/OpenTelemetry/Metrics/MessagingMetricsFeature.cs:37]: Do we need to allocate a list?
```suggestion
        context.Pipeline.OnReceivePipelineCompleted((pipeline, token) => Task.WhenAll(
            processingTimeMetrics.Record(pipeline, token),
            criticalTimeMetrics.Record(pipeline, token)
        );
```

**Review by DavidBoike — APPROVED** [review]

**Comment by mauroservienti** [comment]
Reverting to draft to signal the dependency on https://github.com/Particular/NServiceBus/pull/7077 and preventing accidental merge

File: src/NServiceBus.Core/OpenTelemetry/Metrics/Meters.cs [hunk 2]
```diff
@@ -22,4 +22,7 @@ class Meters
 
     internal static readonly Histogram<double> CriticalTime =
         NServiceBusMeter.CreateHistogram<double>(Metrics.CriticalTime, "s", "The time in seconds between when the message was sent until processed by the endpoint.");
+
+    internal static readonly Histogram<double> ProcessingTime =
```
[the remark this answers is not in the document]
Comment (danielmarbach) [reply src/NServiceBus.Core/OpenTelemetry/Metrics/Meters.cs:26]: Yeah, maybe that's a good idea. But You are welcome to ignore the input.

**Comment by mauroservienti** [comment]
@lailabougria @SzymonPobiega, this is ready for a review

File: src/NServiceBus.AcceptanceTests/Core/OpenTelemetry/When_incoming_message_handled.cs [hunk 3]
```diff
@@ -21,6 +22,14 @@ public async Task Should_record_critical_time()
         AssertMandatoryTags(metricsListener, CriticalTimeMetricName, typeof(MyMessage));
     }
 
+    [Test]
```
Comment (lailabougria) [inline src/NServiceBus.AcceptanceTests/Core/OpenTelemetry/When_incoming_message_handled.cs:25]: Do we need a test that this shouldn't be recorded when handling fails?

**Review by DavidBoike — APPROVED** [review]

**Comment by SzymonPobiega** [comment]
Replaced by https://github.com/Particular/NServiceBus/pull/7097

**Pull request closed without merging**